src/Mailgun/Api/Pagination.php

```diff
@@ -22,12 +23,52 @@
     abstract protected function safeDeserialize(ResponseInterface $response, $className);
 
     /**
+     * @param mixed $response
+     *
+     * @return mixed|null
```

Comment:
Example:
When we call the API endpoint then the SDK should return a Response, right?

What I mean - ideally - we should type hint the expected content, and `mixed` has no sense to me when I am using a SDK that is managing an API resource. From my POV a SDK should return always known resources, no matter if it is an object, an array, a string...